dashboard/test/models/concerns/pd/form_test.rb

```diff
@@ -2,9 +2,39 @@
 
 class DummyForm
   include ActiveModel::Validations
+  include ActiveModel::Dirty
   include Pd::Form
 
+  # minimal implementation from
+  # http://api.rubyonrails.org/classes/ActiveModel/Validations.html
   attr_accessor :form_data
+
+  # minimal implementation from
+  # http://api.rubyonrails.org/classes/ActiveModel/Dirty.html
+  define_attribute_methods :form_data
+
+  def initialize
+    @form_data = nil
+  end
+
+  def form_data=(val)
+    form_data_will_change! unless val == @form_data
+    @form_data = val
+  end
+
+  def save
+    # do persistence work
+    changes_applied
+  end
+
+  def reload!
+    # get the values from the persistence layer
+    clear_changes_information
+  end
+
+  def rollback!
+    restore_attributes
+  end
```

Comment:
But there's no harm right? Unit tests are run on the test environment (db) and also in a transaction so they won't actually leave anything behind. I think it would make the below test more readable.